### PR TITLE
Give testing some love

### PR DESCRIPTION
**Pull request by nicolas-grekas**
`./phpunit` FTW + some goodies

### CODE REVIEW

File: src/Util/TestListenerTrait.php [hunk 1]
```diff
@@ -123,6 +123,10 @@ function {$f['name']}{$f['signature']}
                     $polyfillSignature = ReflectionCaster::castFunctionAbstract(new \ReflectionFunction($testNamespace.'\\'.$f['name']), [], new Stub(), true);
                     $polyfillSignature = ReflectionCaster::getSignature($polyfillSignature);
 
+                    if ('mb_get_info' === $r->name && false === strpos($originalSignature, '|null') && false !== strpos($polyfillSignature, '|null')) {
```
Comment (stof) [inline src/Util/TestListenerTrait.php:122]: can you add a comment about which PHP versions are concerned by this signature patch, to allow cleaning it in the future ?

Comment (nicolas-grekas) [reply src/Util/TestListenerTrait.php:122]: comment added

**Comment by nicolas-grekas** [comment]
ValueError on PHP 8.4 is a regression, reported at https://github.com/php/php-src/issues/15824

**Comment by nicolas-grekas** [comment]
Tests fixed, they show some issues we now can wonder about.
Help wanted @IonBazan @xabbuh @alexandre-daubois or anyone else really :)

**Pull request merged**